tools/src/main/java/org/apache/kafka/tools/consumer/group/ShareGroupCommand.java

```diff
@@ -208,6 +210,50 @@ public void describeGroups() throws ExecutionException, InterruptedException {
             }
         }
 
+        Map<String, Throwable> deleteShareGroups() {
+            List<String> groupIds = opts.options.has(opts.allGroupsOpt)
+                ? listShareGroups()
+                : opts.options.valuesOf(opts.groupOpt);
+
```

Comment:
Because `deleteShareGroups` actually deletes all types of group, I think it would be prudent for this method to check that the groups being deleted are share groups.

In the `--all-groups` case, you're listing the share groups. If you listed the share groups in the other case too, you'd easily be able to see whether the groups are share groups and then avoid deleting consumer groups unawares. Do you think this would be a sensible precaution? It's quite a limited code change.